tests/apichecks/crds_test.go

```diff
@@ -233,6 +233,93 @@ func TestCRDsAcronyms(t *testing.T) {
 	test.CompareGoldenFile(t, "testdata/exceptions/acronyms.txt", want)
 }
 
+// Avoid passing sensitive data as plain text in the CRD
+func TestNoSensitiveField(t *testing.T) {
+	crds, err := crdloader.LoadAllCRDs()
+	if err != nil {
+		t.Fatalf("error loading crds: %v", err)
+	}
+
+	var errs []string
+
+	sensitiveKeywords := []string{
+		"password",
+	}
+	filter := func(fieldPath string) bool {
+		isSensitive := false
+		fieldPath = strings.ToLower(fieldPath)
+		for _, sensitiveWord := range sensitiveKeywords {
+			if strings.HasSuffix(fieldPath, sensitiveWord) {
+				isSensitive = true
+				break
+			}
+		}
+		return isSensitive
+	}
+	for _, crd := range crds {
+
+		for _, version := range crd.Spec.Versions {
+			skepticalfieldPaths := []string{}
+			visitCRDVersion(version, func(field *CRDField) {
+				fieldPath := field.FieldPath
+				isSensitiveSkeptical := false
+				field.FieldPath = strings.ToLower(field.FieldPath)
+				for _, sensitiveWord := range sensitiveKeywords {
+					if strings.Contains(field.FieldPath, sensitiveWord) {
+						isSensitiveSkeptical = true
+						break
+					}
+				}
+				if isSensitiveSkeptical {
+					skepticalfieldPaths = append(skepticalfieldPaths, fieldPath)
+				}
+			})
+			if len(skepticalfieldPaths) == 0 {
+				continue
+			}
+			if len(skepticalfieldPaths) == 1 {
+				if filter(skepticalfieldPaths[0]) {
+					errs = append(errs, fmt.Sprintf("crd=%s version=%v: field %q is sensitive data, should use secretRef", crd.Name, version.Name, skepticalfieldPaths[0]))
+				}
+				continue
+			}
+			sort.Strings(skepticalfieldPaths)
```

Comment:
I think using a set is probably easier?

Reply:
Thank you very much! I really like the new code style!